procedure: CRW 3.0 Story #5a UG Reviewing pull requests (#2244)
codenvy/che-docs: modules/user-guide/partials/benefits-of-PR-review-in-Che.adoc

> :_content-type: PROCEDURE

Reviewer: I'd say the first choice I'd have to make when setting out to write a module about a topic is to decide on the type of module: PROCEDURE, CONCEPT, or REFERENCE.
https://redhat-documentation.github.io/modular-docs/#_creating_modules
In this particular case, I'd try to figure out whether this is better written as a PROCEDURE module or as a CONCEPT module. The CONCEPT is descriptive writing offers more freedom (and can even have subsections!). The PROCEDURE is basically a strict list of imperative sentences that the intended-audience reader can easily perform to achieve a specific, verifiable outcome.

[id="title_{context}"]

> = Benefits of reviewing pull and merge requests in {prod}

Reviewer: Depending on which module type you've chosen:
- The heading of a procedure module starts with a gerund, as per the template: `Doing procedure A`
- The heading of a concept or reference module is a noun phrase and not a gerund.

[role="_abstract"]

> {prod} workspace contains all tools you need to review pull and merge requests from start to finish. By clicking a {prod-short} link, you get access to {prod}-supported web IDE with a ready-to-use workspace where you can run a linter, unit tests, the build and more.

Reviewer 2: Does adding these articles help?
```suggestion
The {prod} workspace contains all tools you need to review pull and merge requests from start to finish. By clicking a {prod-short} link, you get access to the {prod}-supported web IDE with a ready-to-use workspace where you can run a linter, unit tests, the build, and more.
```

.Prerequisites
* You have access to the repository hosted by your Git provider.
* You use a {prod}-supported browser: Google Chrome or Mozilla Firefox.
* You have access to a {prod-short} instance.

.Procedure

> . Open the feature branch to review in {prod-short}. A clone of the branch opens in a workspace with tools for debugging and testing.

Reviewer 2: ```suggestion
. Open the feature branch to review it in {prod-short}. A clone of the branch opens in a workspace with tools for debugging and testing.
```

. Check the pull or merge request changes.
. Run your desired debugging and testing tools:
** Run a linter.
** Run unit tests.
** Run the build.
** Run the application to check for problems.

> . Navigate to UI of your Git provider to leave comment and pull or merge your assigned request.

Reviewer 2: ```suggestion
. Navigate to the UI of your Git provider to leave comments and pull or merge your assigned request.
```

> .Verification

Reviewer: From what I learned, `.Verification` as a separate section is optional.
So if the verification can be contained to one short step, it can be alternatively added as the last step to the procedure. For example as `. Verify that [something is somewhere somehow].`
So if you don't need a separate `.Verification` section, just delete the template lines for it.

> . (optional) Open a second workspace using the main branch of the repository to reproduce a problem.

Reviewer: ```suggestion
. Optional: Open a second workspace using the main branch of the repository to reproduce a problem.
```
This is a tip I learned from the OpenShift docs team. Any optional procedure steps they mark as `. Optional: [Start the sentence with a capital letter]`

> [role="_additional-resources"]
> .Additional resources

Reviewer: ```suggestion
```
Same as for an unused `.Verification` section, just delete the template lines for it.
